movement/movement.h

```diff
@@ -61,7 +61,8 @@ typedef union {
         // altimeter to display feet or meters as easily as it tells a thermometer to display degrees in F or C.
         bool clock_mode_24h : 1;            // indicates whether clock should use 12 or 24 hour mode.
         bool use_imperial_units : 1;        // indicates whether to use metric units (the default) or imperial.
-        uint8_t reserved : 7;               // room for more preferences if needed.
+        bool alarm_enabled : 1;             // indicates wheter there is at least one alarm enabled.
```

Comment:
Typo: wheter.

It would be good if multiple faces were able to interact on this since it's in settings (personally, I would like countdown to play an alarm as well, and it uses an indicator on its own face), but I can't think of a particularly nice way to do it in the current movement architecture (since who is responsible for turning the indicator off?). One simple change would be to use more bits here, and then you could maintain a counter (i.e. +1 for each alarm, -1 for each alarm turned off).

If we were refactoring more (probably not in this PR?), a different approach would be to maintain the alarm logic in movement.c (e.g. movement_schedule_alarm(time, rounds, alarm_note) or movement_schedule_background_task(time, alarm_indicator=true)), which would separate things a bit more cleanly. It wouldn't handle repeating alarms; you'd just do another movement_schedule_background_task when the previous one fired.

Reply:
Thanks a lot for the detailed review!

> Typo: wheter

Will be fixed in next commit.

> ... One simple change would be to use more bits here, and then you could maintain a counter (i.e. +1 for each alarm, -1 for each alarm turned off).

This seems like a good idea at first glance, but I'm a bit conflicted about this. On the one hand, a counter like this would solve situations where you have more than one watch face wanting to use the bell indicator on the clock face. To implement this, you would probably need at least 4 bits for an alarm counter. It would be helpful (and to prevent overflows) to add a dedicated function in `movement.c` taking care of incrementing and decrementing the alarm counter, e.g. `movement_add_alarm(int8_t number_of_alarms)`. 
Also: If you would want to be able to not only _show_ if alarms are set on the clock face, but also _control_ if they fire (like on the original Casio firmware), you would still need the additional `alarm_enabled` flag.
This, on the other hand, feels like eating up a good chunk of reserved bits and adding complexity to movement.c for very specific use case functionality. Maybe @josecastillo has an opinion on this?

> If we were refactoring more (probably not in this PR?), a different approach would be to maintain the alarm logic in movement.c (e.g. movement_schedule_alarm(time, rounds, alarm_note) or movement_schedule_background_task(time, alarm_indicator=true)), which would separate things a bit more cleanly. It wouldn't handle repeating alarms; you'd just do another movement_schedule_background_task when the previous one fired.

If I am not mistaken the `movement_schedule_background_task(time, alarm_indicator=true))` approach would not really be suitable for alarms that can be scheduled for an arbitrary point in time. @joeycastillo wrote this on discord:

> If you implement the _wants_background_task function in your watch face, you'll get that call once a minute whether in sleep mode or wake mode. this is the preferred method for implementing alarms that fire on the minute mark, or scheduled tasks like a sensor reading or calculation. If your watch face schedules a background task with movement_schedule_background_task, the watch will remain in wake mode until the task fires, so that it can check every second for a match. this is the preferred method for timers or countdowns that may not fire exactly at the top of the minute.

Reply:
> If I am not mistaken the movement_schedule_background_task(time, alarm_indicator=true)) approach would not really be suitable for alarms that can be scheduled for an arbitrary point in time.

Sorry, what I forgot to mention (because apparently it was all going on in my head!) is that it would be reasonably easy to change movement_schedule_background_task to check on the sleep mode loop as well, so that what you would have is 1 minute granularity rather than 1 second granularity once the watch has gone to sleep. TBH, I think the current scheduled task implementation is quite prone to footguns, because if ever you register a scheduled task and sleep before it executes, you can get it stuck in there forever (I think, @joeycastillo ?).